### PR TITLE
Add procedure argument check for NULL value

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Add procedure argument check for NULL value. Added test cases for respective procedure class.

## Release notes

(X) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/connector/system/KillQueryProcedure.java [hunk 1]
```diff
@@ -60,6 +61,9 @@ public KillQueryProcedure(Optional<DispatchManager> dispatchManager, AccessContr
     @UsedByGeneratedCode
     public void killQuery(String queryId, String message, ConnectorSession session)
     {
+        checkProcedureArgument(queryId != null, "query_id cannot be null");
+        checkProcedureArgument(message != null, "message cannot be null");
```
Comment (alexjo2144) [inline core/trino-main/src/main/java/io/trino/connector/system/KillQueryProcedure.java:65]: Looks like `message` can be null: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/connector/system/KillQueryProcedure.java#L104

Comment (krvikash) [reply core/trino-main/src/main/java/io/trino/connector/system/KillQueryProcedure.java:65]: Aah, I see. Then should we make message an optional field?

https://github.com/trinodb/trino/blob/91b49be0736be254bbc360cbe70899deac25720c/core/trino-main/src/main/java/io/trino/connector/system/KillQueryProcedure.java#L96

Comment (alexjo2144) [reply core/trino-main/src/main/java/io/trino/connector/system/KillQueryProcedure.java:65]: Yeah, I don't see why it shouldn't be optional.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/DropStatsProcedure.java [hunk 2]
```diff
@@ -96,6 +97,9 @@ public void dropStats(ConnectorSession session, ConnectorAccessControl accessCon
 
     private void doDropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
+        checkProcedureArgument(schema != null, "schema cannot be null");
+        checkProcedureArgument(table != null, "table cannot be null");
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/DropStatsProcedure.java:101]: The argument names are `schema_name` and `table_name`.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/RegisterPartitionProcedure.java [hunk 3]
```diff
@@ -103,6 +104,11 @@ public void registerPartition(ConnectorSession session, ConnectorAccessControl a
 
     private void doRegisterPartition(ConnectorSession session, ConnectorAccessControl accessControl, String schemaName, String tableName, List<String> partitionColumn, List<String> partitionValues, String location)
     {
+        checkProcedureArgument(schemaName != null, "schema_name cannot be null");
+        checkProcedureArgument(tableName != null, "table_name cannot be null");
+        checkProcedureArgument(partitionColumn != null, "partition_columns cannot be null");
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/RegisterPartitionProcedure.java:109]: Could you rename `partitionColumn` to `partitionColumns` in a separated commit or PR?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/UnregisterPartitionProcedure.java [hunk 4]
```diff
@@ -87,6 +88,11 @@ public void unregisterPartition(ConnectorSession session, ConnectorAccessControl
 
     private void doUnregisterPartition(ConnectorSession session, ConnectorAccessControl accessControl, String schemaName, String tableName, List<String> partitionColumn, List<String> partitionValues)
     {
+        checkProcedureArgument(schemaName != null, "schema_name cannot be null");
+        checkProcedureArgument(tableName != null, "table_name cannot be null");
+        checkProcedureArgument(partitionColumn != null, "partition_columns cannot be null");
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/UnregisterPartitionProcedure.java:93]: Could you rename `partitionColumn` to `partitionColumns` in a separated commit or PR?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHivePartitionProcedures.java [hunk 5]
```diff
@@ -122,6 +123,19 @@ public void testUnregisterMissingPartitionShouldFail()
                 .hasMessageContaining("Partition 'col=f' does not exist");
     }
 
+    @Test(groups = {HIVE_PARTITIONING, SMOKE})
+    public void testUnregisterPartitionWithNullArgument()
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHivePartitionProcedures.java:127]: Let's move to around `BaseHiveConnectorTest#testUnregisterRegisterPartition`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHivePartitionProcedures.java [hunk 6]
```diff
@@ -239,6 +253,19 @@ public void testRegisterPartitionFromAnyLocation()
         assertThat(getValues(FIRST_TABLE)).containsOnly(1, 2, 3);
     }
 
+    @Test(groups = {HIVE_PARTITIONING, SMOKE})
+    public void testRegisterPartitionWithNullArgument()
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHivePartitionProcedures.java:257]: diito.

File: testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java [hunk 7]
```diff
@@ -69,6 +69,13 @@ public void tearDown()
     @Test(timeOut = 60_000)
     public void testKillQuery()
             throws Exception
+    {
+        killQuery(Optional.of("because"), "Message: because");
+        killQuery(Optional.empty(), "No message provided.");
+    }
+
+    private void killQuery(Optional<String> argumentMessage, String expectedKilledMessage)
```
Comment (ebyhr) [inline testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java:77]: How about changing `Optional<String>` to `Function<String, String>`?

File: testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java [hunk 8]
```diff
@@ -69,6 +69,13 @@ public void tearDown()
...
+            throws Exception
```
Comment (ebyhr) [inline testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java:78]: Avoid throwing `Exception`. Throwing `Exception` is fine in test methods, but this is a helper method, not a test method. 

Another approach is using `Uninterruptibles.sleepUninterruptibly`.

File: testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java [hunk 9]
```diff
@@ -97,11 +104,21 @@ public void testKillQuery()
...
                 .isInstanceOf(ExecutionException.class)
-                .hasMessageContaining("Query killed. Message: because");
+                .hasMessageContaining("Query killed. " + expectedKilledMessage);
+    }
+
+    @Test
+    public void testKillQueryWithNullArgument()
+    {
+        assertThatThrownBy(() -> getQueryRunner().execute("CALL system.runtime.kill_query(NULL, 'should fail')"))
+                .hasMessageContaining("query_id cannot be null");
```
Comment (ebyhr) [inline testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java:121]: Use `assertQueryFails`

**Comment by krvikash** [comment]
@alexjo2144, @ebyhr addressed review comments.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/UnregisterPartitionProcedure.java (hunk 4 above)
Comment (ebyhr) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/UnregisterPartitionProcedure.java]: No change is requested. Ideally, the above change should be commited first to reduce diff.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**